samples/csharp/sharedcontent/console/speech_synthesis_samples.cs

```diff
@@ -850,5 +850,72 @@ public static async Task SynthesisGetAvailableVoicesAsync()
+            {
+                synthesisResult = await synthesizer.SpeakTextAsync(text).ConfigureAwait(false);
+            }
+            
+
+            return synthesisResult;
+        }
+
+        // Speech synthesis with backup subscription region.
+        public static async Task SynthesizeOnceToSpeakerAsyncSwitchSecondaryRegion()
+        {
+            // Create a speech resource with primary subscription key and service region.
+            // Also create a speech resource with secondary subscription key and service region
+            SpeechSynthesisResult synthesisResult = await SynthesizeOnceAsyncInternal("PrimarySubscriptionKey", "PrimarySubscriptionRegion" );
+            if (synthesisResult.Reason == ResultReason.Canceled)
+            {
+                SpeechSynthesisCancellationDetails details = SpeechSynthesisCancellationDetails.FromResult(synthesisResult);
+                if (details.ErrorCode == CancellationErrorCode.ConnectionFailure
+                    || details.ErrorCode == CancellationErrorCode.ServiceUnavailable
+                    || details.ErrorCode == CancellationErrorCode.ServiceTimeout)
+                {
+                    synthesisResult = await SynthesizeOnceAsyncInternal("SecondarySubscriptionRegion", "SecondarySubscriptionRegion");
+                }
+            }
+        }
+
+        // Custom voice with backup.
+        // If custom voice service is unavailable, you can fallback to a standard platform voice
+        // or your custom voice service in another region.
+        // This sample uses platform voice as back up.
+        public static async Task SynthesizeOnceUseCustomVoiceToSpeakerAsyncSwitchPlatformVoice()
+        {
+            // Create a speech resource with primary subscription key and service region.
+            // Also create a speech resource with secondary subscription key and service region
+            SpeechSynthesisResult synthesisResult = await SynthesizeOnceAsyncInternal("PrimarySubscriptionKey",
+                "PrimarySubscriptionRegion", "YourEndpointId", "YourVoiceName");
+            if (synthesisResult.Reason == ResultReason.Canceled)
+            {
+                SpeechSynthesisCancellationDetails details = SpeechSynthesisCancellationDetails.FromResult(synthesisResult);
+                if (details.ErrorCode == CancellationErrorCode.ConnectionFailure
+                    || details.ErrorCode == CancellationErrorCode.ServiceUnavailable
+                    || details.ErrorCode == CancellationErrorCode.ServiceTimeout 
+                    || details.ErrorDetails.Contains("Error code: 1007"))
+                {
+                    // Synthesize using platform voice. The SecondarySubscription could be same with primary one
+                    synthesisResult = await SynthesizeOnceAsyncInternal("SecondarySubscriptionRegion", "SecondarySubscriptionRegion");
+                }
+            }
+        }
     }
 }
```

Comment:
Please add a separate sample for Custom Voice fallback to Custom Voice.
        // Custom voice with backup.
        // If custom voice service is unavailable, you can fallback to custom voice service in another region.
        public static async Task SynthesizeOnceUseCustomVoiceToSpeakerAsyncSwitchSecondaryRegion()
        {
            // Create a speech resource with primary subscription key and service region.
            // Also create a speech resource with secondary subscription key and service region.
            // Copy Custom Voice model from primary region to secondary region and deploy.
            SpeechSynthesisResult synthesisResult = await SynthesizeOnceAsyncInternal("PrimarySubscriptionKey",
                "PrimarySubscriptionRegion", "YourEndpointIdOnPrimaryRegion", "YourVoiceName");
            if (synthesisResult.Reason == ResultReason.Canceled)
            {
                SpeechSynthesisCancellationDetails details = SpeechSynthesisCancellationDetails.FromResult(synthesisResult);
                if (details.ErrorCode == CancellationErrorCode.ConnectionFailure
                    || details.ErrorCode == CancellationErrorCode.ServiceUnavailable
                    || details.ErrorCode == CancellationErrorCode.ServiceTimeout
                    || details.ErrorDetails.Contains("Error code: 1007"))
                {
                    // Synthesize using platform voice. The SecondarySubscription could be same with primary one
                    synthesisResult = await SynthesizeOnceAsyncInternal("SecondarySubscriptionRegion", "SecondarySubscriptionRegion",
                        "YourEndpointIdOnSecondaryRegion", "YourVoiceName");
                }
            }
        }

Reply:
added, thx